File: go/0083_remove_duplicates_from_sorted_list.go
package main

func deleteDuplicates(head *ListNode) *ListNode {
	if head == nil {
		return nil
	}
	dummyHead := head

	for head.Next != nil {
		if head.Val == head.Next.Val {
			head.Next = head.Next.Next
			continue
		}
		head = head.Next
	}
	return dummyHead
}

File: python/0083_remove_duplicates_from_sorted_list.py
class Solution:
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        cur = head
        while cur is not None:
            while cur.next is not None and cur.val == cur.next.val:
                cur.next = cur.next.next
            cur = cur.next
        return head